apps/jobs/views.py

```diff
@@ -745,3 +749,354 @@ def update_submission(request, challenge_pk):
+        else:
+            logger.info("No new submission is received!")
+            message_receipt_handle = None
+            message_body = None
+
+        response_data = {
+            'message_body': message_body,
+            'message_receipt_handle': message_receipt_handle
+        }
+        return Response(response_data, status=status.HTTP_200_OK)
+    except botocore.exceptions.ClientError as ex:
+        response_data = ex
+        logger.exception("The following exception is raised {}".format(ex))
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET'])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def delete_submission_message_by_queue_name(request, queue_name, receipt_handle):
+    """
+    API endpoint to delete the submission message from the AWS SQS Queue
+    """
+    try:
+        challenge = Challenge.objects.get(queue=queue_name)
+    except Challenge.DoesNotExist:
+        response_data = {
+            'error': 'Challenge with queue name {} does not exists!'.format(queue_name)
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge_pk = challenge.id
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            'error': 'Sorry, you are not authorized to access this resource!'
+        }
+        return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
+
+    if settings.DEBUG or settings.TEST:
+        queue_name = 'evalai_submission_queue'
+        sqs = boto3.resource('sqs',
+                             endpoint_url=os.environ.get('AWS_SQS_ENDPOINT', 'http://sqs:9324'),
+                             region_name=os.environ.get('AWS_DEFAULT_REGION', 'us-east-1'),
+                             aws_secret_access_key=os.environ.get('AWS_SECRET_ACCESS_KEY', 'x'),
+                             aws_access_key_id=os.environ.get('AWS_ACCESS_KEY_ID', 'x'),)
+    else:
+        sqs = boto3.resource('sqs',
+                             region_name=os.environ.get('AWS_DEFAULT_REGION', 'us-east-1'),
+                             aws_secret_access_key=os.environ.get('AWS_SECRET_ACCESS_KEY'),
+                             aws_access_key_id=os.environ.get('AWS_ACCESS_KEY_ID'),)
+    # Check if the queue exists. If no, then create one
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+    except botocore.exceptions.ClientError as ex:
+        if ex.response['Error']['Code'] != 'AWS.SimpleQueueService.NonExistentQueue':
+            logger.exception('Cannot get queue: {}'.format(queue_name))
+
+    try:
+        message = queue.Message(receipt_handle)
+        message.delete()
+        response_data = {
+            'success': 'The message is successfully deleted from the queue!'
```

Comment:
Drop `!` from everywhere

Reply:
Can we please change the message `'Message deleted successfully from the queue: {}'format(queue_name)`